### PR TITLE
[Sui-test-validator] adding persisted state option

### PR DESCRIPTION
**Pull request by healthydeve**
## Description 

Adding a persisted state on the ./sui-test-validator if we choose to pass in a network config

The current configuration will save a file in the .sui/.sui_config/.sui_cluster_test_config/ path of your local directory. We are able to run locally and have a saved state with this.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-config/src/genesis_config.rs [hunk 1]
```diff
@@ -230,6 +230,19 @@ impl GenesisConfig {
...
 
+    pub fn for_local_testing_with_addresses_and_faucet(addresses: Vec<SuiAddress>) -> Self {
+        let mut custom_genesis = Self::custom_genesis_with_addresses(
+            DEFAULT_NUMBER_OF_AUTHORITIES,
+            addresses,
+            DEFAULT_NUMBER_OF_OBJECT_PER_ACCOUNT,
+        );
+        custom_genesis.accounts.push(AccountConfig {
+            address: None,
+            gas_amounts: vec![DEFAULT_GAS_AMOUNT; DEFAULT_NUMBER_OF_OBJECT_PER_ACCOUNT],
```
Comment (wlmyng) [inline crates/sui-config/src/genesis_config.rs:241]: what is this for?

**Comment by aji7777** [comment]
I not list help me
In to list
Email: kanankiri151998@gmail.com
My.id. 62910930

File: crates/sui-config/src/genesis_config.rs (hunk 1 above)
Comment (healthydeve) [reply crates/sui-config/src/genesis_config.rs:241]: It creates the genesis config that is used for loading the validator nodes and various other set ups.

Comment (healthydeve) [reply crates/sui-config/src/genesis_config.rs]: Removed it, it was unnecessary

File: crates/sui-cluster-test/src/cluster.rs [hunk 2]
```diff
@@ -162,10 +168,26 @@ impl LocalNewCluster {
...
+            (
+                Some(sui_cluster_test_config_dir()?.join(SUI_NETWORK_CONFIG)),
+                true,
+            )
+        } else {
+            (None, false)
+        };
+        // Let the faucet account hold 100 gas objects on genesis and we reuse existing config
+        // if it was provided.
+        let genesis_config = if let Some(config) = genesis_config {
```
Comment (wlmyng) [inline crates/sui-cluster-test/src/cluster.rs:186]: suggestion: `let genesis_config = genesis_config.unwrap_or_else(|| GenesisConfig::custom_genesis(4, 1, 100));`

**Review by wlmyng — APPROVED** [review]

**Pull request merged**